packages/analytics-client-common/src/attribution/campaign-tracker.ts

```diff
@@ -42,17 +42,24 @@ export class CampaignTracker implements ICampaignTracker {
     this.initialEmptyValue = options.initialEmptyValue ?? EMPTY_VALUE;
   }
 
-  isNewCampaign(current: Campaign, previous: Campaign | undefined) {
-    const { referrer: _referrer, ...currentCampaign } = current;
-    const { referrer: _previous_referrer, ...previousCampaign } = previous || {};
+  isNewCampaign(
+    current: Campaign,
+    previous: Campaign | undefined,
+    removeSubdomainsWhenComparingReferringDomains = false,
+  ) {
+    const { referrer, referring_domain, ...currentCampaign } = current;
+    const { referrer: _previous_referrer, referring_domain: prevReferringDomain, ...previousCampaign } = previous || {};
```

Comment:
Deconstructing `referring_domain: prevReferringDomain` will remove this key/value pair from `previousCampaign` and will not be part of the `hasNewCampaign` check. I think this will cause `hasNewCampaign` to always be true since the two objects do not contain the same key/value pair entries.

Reply:
referring_domain is deconstructed from both current and previous campaign objects, so I wouldn't expect it to cause a problem. Am I misreading this?

Reply:
Ah no i misread it, i didn't notice `referring_domain` deconstructed from current.